chain/block.go

```diff
@@ -894,6 +896,79 @@ func (b *StatefulBlock) FeeManager() *fees.Manager {
 	return b.feeManager
 }
 
+func (b *StatefulBlock) ExecutedBlock() *ExecutedBlock {
+	return &ExecutedBlock{
+		StatelessBlock: b.StatelessBlock,
+		Results:        b.results,
+		UnitPrices:     b.feeManager.UnitPrices(),
+		blockBytes:     b.Bytes(),
+		id:             b.ID(),
+	}
+}
+
+type ExecutedBlock struct {
+	*StatelessBlock `json:"block"`
+	Results         []*Result             `json:"results"`
+	UnitPrices      publicfees.Dimensions `json:"unitPrices"`
+
+	blockBytes []byte
+	id         ids.ID
+}
+
+func (b *ExecutedBlock) ID() ids.ID { return b.id }
+
+func (b *ExecutedBlock) Marshal() ([]byte, error) {
+	size := codec.BytesLen(b.blockBytes) + codec.CummSize(b.Results) + publicfees.DimensionsLen
+	writer := codec.NewWriter(size, consts.NetworkSizeLimit)
+
+	writer.PackBytes(b.blockBytes)
+	resultBytes, err := MarshalResults(b.Results)
+	if err != nil {
+		return nil, err
+	}
+	writer.PackBytes(resultBytes)
+	writer.PackFixedBytes(b.UnitPrices.Bytes())
+
+	return writer.Bytes(), writer.Err()
+}
+
+func UnmarshalExecutedBlock(bytes []byte, parser Parser) (*ExecutedBlock, error) {
```

Comment:
nit: should `Unmarshal` have a receiver on the struct too (or `Marshal` be static)? Seems weird that it's not consistent.

Reply:
Following current style here, happy to discuss for a separate change

Reply:
This is new code introduced as part of the diff... would it be reasonable to do it as part of this PR?

Reply:
All of the block marshalling / unmarshalling code is consistent with this change, so I'd rather not introduce a new style for one function.

Reply:
I'm also not sure moving it to a function on the pointer receiver is better. Either way, I don't think it belongs in this PR.